src/udev/udev-node.c

```diff
@@ -588,10 +593,7 @@ int udev_node_remove(sd_device *dev) {
                 return log_device_debug_errno(dev, r, "Failed to get device path: %m");
 
         /* remove /dev/{block,char}/$major:$minor */
-        if (unlink(filename) < 0 && errno != ENOENT)
-                return log_device_debug_errno(dev, errno, "Failed to remove '%s': %m", filename);
-
-        return 0;
+        return node_remove_symlink(dev, filename);
```

Comment:
this wasn't calling rmdir_parents before, is this intended?

Reply:
Yes. I do not think it is not necessary to keep /dev/block/ or /dev/char/ if empty.